File: tests/tests/fixtures/python_repo/services/payment_service.py
"""services/payment_service.py — fixture for DocWiki tests."""

from __future__ import annotations

from repositories.payment_repository import PaymentRepository


class PaymentService:
    """Business logic for payment processing."""

    def __init__(self, repo: PaymentRepository) -> None:
        self._repo = repo

    def charge(self, order, amount) -> str:
        """Charge the order and persist the record."""
        record = self._repo.save(order)  # self call resolved
        return self._confirm(record)

    def refund(self, order_id: str) -> str:
        rec = self._repo.find(order_id)
        return self._confirm(rec)

    def _confirm(self, record) -> str:
        return f"confirmed:{record}"

File: tests/tests/fixtures/java_repo/src/main/java/com/example/service/PaymentService.java
package com.example.service;

import com.example.repository.PaymentRepository;

public class PaymentService implements Chargeable {
    private final PaymentRepository repo;

    public PaymentService(PaymentRepository repo) {
        this.repo = repo;
    }

    @Override
    public String charge(Order order, double amount) {
        String record = repo.save(order);
        return this.confirm(record);
    }

    public String charge(Order order) {
        return this.charge(order, order.getAmount());
    }

    public String refund(String orderId) {
        String rec = repo.find(orderId);
        return this.confirm(rec);
    }

    private String confirm(String record) {
        return "confirmed:" + record;
    }
}
